plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/expression/RewriteLike.java

```diff
@@ -0,0 +1,103 @@
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.type;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.variable;
+import static io.trino.plugin.clickhouse.ClickHouseClient.supportsPushdown;
+import static io.trino.spi.expression.StandardFunctions.LIKE_FUNCTION_NAME;
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class RewriteLike
+        implements ConnectorExpressionRule<Call, ParameterizedExpression>
+{
+    private static final Capture<Variable> LIKE_VALUE = newCapture();
+    // TODO allow Variable as a LIKE_PATTERN: "SELECT * FROM t WHERE column_a LIKE column_b" is a valid query in ClickHouse
+    // only Constant is allowed as LIKE_PATTERN, because according to
+    // https://clickhouse.com/docs/en/sql-reference/functions/string-search-functions#like
+    // ClickHouse requires backslashes in strings to be quoted as well, so you would actually need to write \\%, \\_ and \\\\ to match against literal %, _ and \
+    // if "column_a LIKE column_b" is pushed down, it requires more thorough consideration how to process escaping.
+    private static final Capture<Constant> LIKE_PATTERN = newCapture();
+    private static final Pattern<Call> PATTERN = call()
+            .with(functionName().equalTo(LIKE_FUNCTION_NAME))
+            .with(type().equalTo(BOOLEAN))
+            .with(argumentCount().equalTo(2))
+            .with(argument(0).matching(variable()
+                    .with(type().matching(type -> type instanceof CharType || type instanceof VarcharType))
+                    .matching((Variable variable, RewriteContext<ParameterizedExpression> context) -> supportsPushdown(variable, context))
+                    .capturedAs(LIKE_VALUE)))
+            .with(argument(1).matching(constant()
+                    .with(type().matching(type -> type instanceof CharType || type instanceof VarcharType))
+                    .capturedAs(LIKE_PATTERN)));
+
+    @Override
+    public Pattern<Call> getPattern()
+    {
+        return PATTERN;
+    }
+
+    @Override
+    public Optional<ParameterizedExpression> rewrite(Call expression, Captures captures, RewriteContext<ParameterizedExpression> context)
+    {
+        Optional<ParameterizedExpression> value = context.defaultRewrite(captures.get(LIKE_VALUE));
+        if (value.isEmpty()) {
+            return Optional.empty();
+        }
+        Optional<ParameterizedExpression> pattern = context.defaultRewrite(captures.get(LIKE_PATTERN));
+        if (pattern.isEmpty()) {
+            return Optional.empty();
+        }
+
+        // Capture<Constant> LIKE_PATTERN guarantees that value is a single varchar
+        QueryParameter patternParameter = pattern.get().parameters().getFirst();
+        Slice slice = (Slice) patternParameter.getValue().orElseThrow();
```

Comment:
Trino could have the Like pattern as `Null` - Can we add test cases for them as well ?

Reply:
Instead of handling them on the `QueryParameter` - Can we apply them directly on the `Constant` - i.e via 

```
Constant likePattern = captures.get(LIKE_PATTERN);
if (likePattern == null || ((Slice) likePattern.getValue()).toStringUtf8().contains("\\") {
            return Optional.empty();
}
```

Reply:
Can we have an additional check for null with LikePattern - it would be replaced as empty values node - just to ensure that we don't break them or introduce new regression in the future.